compute_modules/annotations.py

```diff
@@ -19,9 +19,12 @@
 from .startup import start_compute_module
 
 
-def function(func: Callable[..., Any]) -> Callable[..., Any]:
-    add_function(func)
-    return func
+def function(streaming: bool = False) -> Callable[..., Any]:
+    def function_without_params(func: Callable[..., Any]) -> Callable[..., Any]:
+        add_function(func, streaming)
+        return func
+
+    return function_without_params
```

Comment:
To avoid breaking backwards compatibility I'm thinking maybe we should [allow with or without parentthesis](https://stackoverflow.com/questions/35572663/using-python-decorator-with-or-without-parentheses/35572746#35572746). 

This should do the trick
```python
def function(
    maybe_func: Any = None,
    *,
    streaming: bool = False,
) -> Callable[..., Any]:
    def function_without_params(func: Callable[..., Any]) -> Callable[..., Any]:
        add_function(func, streaming)
        return func

    if callable(maybe_func):
        return function_without_params(maybe_func)
    return function_without_params

```

Reply:
Thanks! Updated